### PR TITLE
Quickstart updates

### PR DESCRIPTION
**Pull request by MeghaB**
Explicitly specific activity for callback binding, use PhoneAuthOptions

### CODE REVIEW

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/java/PhoneAuthActivity.java [hunk 1]
```diff
@@ -175,12 +176,16 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
...
-                mCallbacks);        // OnVerificationStateChangedCallbacks
+
+        PhoneAuthOptions options = 
+          PhoneAuthOptions.newBuilder(mAuth) 
+              .setPhoneNumber(phoneNumber)       // Phone number to verify
+              .setTimeout(60L, TimeUnit.SECONDS) // Timeout and unit
+              .setActivity(this)                 // Activity (for callback binding)
+              .setCallbacks(mCallbacks)          // OnVerificationStateChangedCallbacks
+              .build();
+          PhoneAuthProvider.getInstance()
```
Comment (malcolmdeck) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/java/PhoneAuthActivity.java:187]: You don't need #getInstance() here - this is gonna break. (My bad, I sent you a bad ref). See the docs:
https://firebase.google.com/docs/reference/android/com/google/firebase/auth/PhoneAuthProvider#verifyPhoneNumber(com.google.firebase.auth.PhoneAuthOptions)

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt [hunk 2]
```diff
@@ -151,12 +152,14 @@ class PhoneAuthActivity : AppCompatActivity(), View.OnClickListener {
...
-                this, // Activity (for callback binding)
-                callbacks) // OnVerificationStateChangedCallbacks
+        val options = 
+          PhoneAuthOptions.newBuilder(mAuth)
+              .setPhoneNumber(phoneNumber)       // Phone number to verify
+              .setTimeout(60L, TimeUnit.SECONDS) // Timeout and unit
+              .setActivity(this)                 // Activity (for callback binding)
+              .setCallbacks(mCallbacks)          // OnVerificationStateChangedCallbacks
+              .build();
+        PhoneAuthProvider.getInstance().verifyPhoneNumber(options);
```
Comment (malcolmdeck) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:162]: Same here

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/java/PhoneAuthActivity.java (hunk 1 above)
Comment (MeghaB) [reply auth/app/src/main/java/com/google/firebase/quickstart/auth/java/PhoneAuthActivity.java:187]: No worries - I assumed it was part of the api changes. Updating accordingly!

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt (hunk 2 above)
Comment (MeghaB) [reply auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:162]: ack

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt [hunk 3]
```diff
@@ -151,12 +152,13 @@ class PhoneAuthActivity : AppCompatActivity(), View.OnClickListener {
...
-                TimeUnit.SECONDS, // Unit of timeout
-                this, // Activity (for callback binding)
-                callbacks) // OnVerificationStateChangedCallbacks
+        val options = PhoneAuthOptions.newBuilder(mAuth)
+              .setPhoneNumber(phoneNumber)       // Phone number to verify
+              .setTimeout(60L, TimeUnit.SECONDS) // Timeout and unit
+              .setActivity(this)                 // Activity (for callback binding)
+              .setCallbacks(mCallbacks)          // OnVerificationStateChangedCallbacks
+              .build();
+        PhoneAuthProvider.verifyPhoneNumber(options);
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:161]:
```suggestion
              .build()
        PhoneAuthProvider.verifyPhoneNumber(options)
```

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt [hunk 4]
```diff
@@ -174,13 +176,14 @@ class PhoneAuthActivity : AppCompatActivity(), View.OnClickListener {
...
-                callbacks, // OnVerificationStateChangedCallbacks
-                token) // ForceResendingToken from callbacks
+        val options = PhoneAuthOptions.newBuilder(mAuth)
+                .setPhoneNumber(phoneNumber)       // Phone number to verify
+                .setTimeout(60L, TimeUnit.SECONDS) // Timeout and unit
+                .setActivity(this)                 // Activity (for callback binding)
+                .setCallbacks(mCallbacks)          // OnVerificationStateChangedCallbacks
+                .setForceResendingToken(token)     // ForceResendingToken from callbacks
+                .build();
+        PhoneAuthProvider.verifyPhoneNumber(options);
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/PhoneAuthActivity.kt:186]:
```suggestion
                .build()
        PhoneAuthProvider.verifyPhoneNumber(options)
```

**Review by rosariopfernandes — COMMENTED** [review]
Changes look good, but I don't think we need these semicolons in Kotlin :wink:

**Pull request merged**